packages/next/build/webpack-config.ts

```diff
@@ -89,6 +90,99 @@ function getOptimizedAliases(isServer: boolean): { [pkg: string]: string } {
   }
 }
 
+async function getPostCssPlugins(dir: string): Promise<unknown[]> {
+  function load(plugins: { [key: string]: object }) {
+    return Object.keys(plugins).map(pkg => {
+      const pluginPath = resolveRequest(pkg, `${dir}/`)
+
+      const options = plugins[pkg]
+      if (options == null || Object.keys(options).length === 0) {
+        return require(pluginPath)
+      }
+      return require(pluginPath)(options)
+    })
+  }
+
+  const config = await loadConfig<{ plugins: { [key: string]: object } }>(
+    dir,
+    'postcss'
+  )
+
+  let target: unknown[]
+
+  if (!config) {
+    target = load({
+      [require.resolve('postcss-flexbugs-fixes')]: {},
+      [require.resolve('postcss-preset-env')]: {
+        autoprefixer: {
+          // Disable legacy flexbox support
+          flexbox: 'no-2009',
+        },
+        // Enable CSS features that have shipped to the
+        // web platform, i.e. in 2+ browsers unflagged.
+        stage: 3,
+      },
+    })
+  } else {
+    const plugins = config.plugins
+    if (plugins == null || typeof plugins !== 'object') {
+      throw new Error(
+        `Your custom PostCSS configuration must export a \`plugins\` key.`
+      )
+    }
+
+    const invalidKey = Object.keys(config).find(key => key !== 'plugins')
+    if (invalidKey) {
+      console.warn(
+        `${chalk.yellow.bold(
+          'Warning'
+        )}: Your PostCSS configuration defines a field which is not supported (\`${invalidKey}\`). ` +
+          `Please remove this configuration value.`
+      )
+    }
+
+    // These plugins cannot be enabled by the user because they'll conflict with
+    // `css-loader`'s behavior to make us compatible with webpack.
+    ;[
+      'postcss-modules-values',
+      'postcss-modules-scope',
+      'postcss-modules-extract-imports',
+      'postcss-modules-local-by-default',
+    ].forEach(plugin => {
+      if (!plugins.hasOwnProperty(plugin)) {
+        return
+      }
+
+      console.warn(
+        `${chalk.yellow.bold('Warning')}: Please remove the ${chalk.underline(
+          plugin
+        )} plugin from your PostCSS configuration. ` +
+          `This plugin is automatically configured by Next.js.`
+      )
+      delete plugins[plugin]
```

Comment:
Should this be a soft error? Seems like we should error hard instead of just removing the plugins and warning 🤔